File: viewCandidato/src/app/model/DadosPessoais.ts

export class DadosPessoais {

    foto: any;
    nomeCompleto: string;
    nascimento: Date;
    rg: string;
    cpf: string;
    nacionalidade: string;
    naturalidade: string;
    estadoCivil: string; // solteiro, casado, divorciado, união estável
    genero: string; // M F

    possuiFilhos: boolean;
    possuiHabilitacao: boolean;
    possuiVeiculo: boolean;
    possuiNecessidadesEspeciais: boolean;
    disponibilidadeMudar: boolean;
    pretensaoSalarial: number;

    linkFace: string;
    linkedIn: string;
    instagram: string;

    logradouro: string;
    bairro: string;
    cidade: string;
    uf: string;
    cep: string;
    complemento: string; // opcional/ 
    
    constructor(){

    }

}

File: server/src/main/java/br/com/samayah/happyJobs/model/DadosPessoais.java
package br.com.samayah.happyJobs.model;

import javax.persistence.Embeddable;
import javax.persistence.Id;

import lombok.Data;
import lombok.NoArgsConstructor;

@Embeddable
@Data
@NoArgsConstructor
public class DadosPessoais {

	private String foto;
	private String nomeCompleto;
	private String nascimento;
	private String rg;
	private String cpf;
	private String nacionalidade;
	private String naturalidade;
	private String estadoCivil; // solteiro, casado, divorciado, união estável
	private String genero; // M F

	private Boolean possuiFilhos;
	private Boolean possuiHabilitacao;
	private Boolean possuiVeiculo;
	private Boolean possuiNecessidadesEspeciais;
	private Boolean disponibilidadeMudar;
	
	private String pretensaoSalarial;

	private String linkFace;
	private String linkedIn;
	private String instagram;
	
	private String logradouro;
	private String bairro;
	private String cidade;
	private String uf;
	private String cep;
	private String complemento; // opcional/ 
	
}
